### ISSUE TITLE
Library size limitations

### ISSUE THREAD

**Issue by pwinnski** [YACReader/yacreader#311]
In YACReaderLibrary, I initially tried importing about 30,000 cbz/cbr files as a single library, and that took many, many, many days, but then the library was unusable. Any attempt to double-click a file would launch YACReader but then say it couldn't communicate with YACReaderLibrary.

So I decided to move one level down, and started to create a separate library for each of the top-level folders. As I hit the plus to create my eleventh library, I got a popup:
![image](https://user-images.githubusercontent.com/48061904/183433021-5c592c58-a9dc-4067-a1c2-19ff1490bc9e.png)

I am fascinated by how long it take to navigate even with ten libraries. Using something like DB Browser for SQLite is lightning-fast, so I'm guessing YACReaderLibrary is trying to load all of the table information in memory, and the memory allocation is taking a long time? 

And now I'm not sure. Is this warning an attempt to educate me about the normal use of YACReaderLibrary, which happens to be unavailable to me due to scale? Or is there going to be a problem if I create 28 or so different libraries, none of which have more than 5139 files?

**Comment by luisangelsm** [comment]
Something is definitely off with your setup. 30000 comics should take 1 hour or so...And navigation, opening content, etc, should be really quick.

Where do you have your comics stored? What machine are you running YACReader on?

About that dialog, yeah, it really tells you that you should put all your comics together like you did the first time.

**Comment by pwinnski** [comment]
The comics are all on a network drive being shared via AFP to an M1 Mac mini with only 8GB of RAM.

Looking at my IM conversations, I actually think the full import took somewhere between 12 and 24 hours, but was then unusable as I described. To read any comic, I had to right-click and choose "Open containing folder..." and then double-click it in the Finder to launch YACReader.

I'm not sure which is impact the speed more: the network share, or the 8GB of RAM on this Mac. Maybe it's a double-whanmny.

**Comment by pwinnski** [comment]
I kicked off another import at 9:17 local time, and I'll keep an eye on YACReaderLibrary's memory usage and the machine's swap, I guess.

**Comment by luisangelsm** [comment]
RAM shouldn't be the issue, you can check in the Activity Monitor how much RAM do you have free and how much YACReaderLibrary is taking.

SQLite is not the fastest thing to use on a network file system, but my tests years ago showed that it should be good enough, especially for reading.

I am not familiar with AFP, maybe there are some settings you can use when mounting the volume in your system. It seems that some people have performance issues: https://forums.macrumors.com/threads/very-slow-access-to-smb-and-afp-shares.2285822

**Comment by pwinnski** [comment]
I remember doing some network speed tests when I first got this machine, but it has been a while. When is import finishes I will dig into file sharing performance. 

Watching the progress of the import it is at 612 coversn the .yacreaderlibrary dir, and memory doesn’t seem to be climbing very quickly, so I suspect you are right and it is all down to file sharing performance. 664 now.

**Comment by pwinnski** [comment]
Thank you, @luisangelsm, you have sent me down the right path. This isn't a YACReaderLibrary issue at all, YACReaderLibrary is merely a victim of my currently-poor file-sharing performance. I'm going to switch to either SMB or NFS (whichever performs better in tests) and tweak the settings before trying an import again. Presumably then I'll get read performance better than 11MB/s(!)

**Comment by pwinnski** [comment]
In case anyone else happens along with a similar issue, I disabled AFP completely and enabled SMB on my NAS (a Synology), and disabled encryption since this is only a local share (firewall blocks sharing remotely), but that still only pushed my speed from 10MB/s to 16MB/s. Then I found [a page that suggested disabling the "delayed ack" setting](http://www.debugginlife.com/solved-very-slow-file-transfer-on-mac-shares-smbafp/) and my speed went from 16MB/s to 2885MB/s, and no, that's not a typo. 

In case that page disappears in the future, it's `sudo sysctl -w net.inet.tcp.delayed_ack=0` for immediate effect, or setting the same value in `/etc/sysctl.conf` to persist through reboots.
